test/TesterInternal/StorageTests/PersistenceGrainTests.cs

```diff
@@ -1200,45 +1177,58 @@ private void SetErrorInjection(string providerName, ErrorInjectionPoint errorInj
                 //}
             }
         }
-
-        private MockStorageProvider FindStorageProviderInUse(string providerName, bool okNull = false)
+        private bool CanGetStorageProviderInUse(string providerName)
         {
-            MockStorageProvider providerInUse = null;
-            SiloHandle siloInUse = null;
-            List<SiloHandle> silos = this.HostedCluster.GetActiveSilos().ToList();
-            foreach (var siloHandle in silos)
+            foreach (var siloHandle in this.HostedCluster.GetActiveSilos())
             {
-                MockStorageProvider provider = (MockStorageProvider)siloHandle.AppDomainTestHook.GetStorageProvider(providerName);
-                Assert.NotNull(provider);
-                if (provider.ReadCount > 0)
+                if (siloHandle.TestHook.CanGetStorageProvider(providerName).Result)
                 {
-                    Assert.Null(providerInUse);
-                   
-                    providerInUse = provider;
-                    siloInUse = siloHandle;
```

Comment:
Right, either remove the first check altogether, or rename it to something like HasStorageProvider, which is effectively what it's asserting